sorting/merge_sort.c

```diff
@@ -97,24 +99,36 @@ void merge_sort(int *a, int n, int l, int r)
 }
 /** @} */
 
-/** Main function */
-int main(void)
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test()
 {
-    int *a, n, i;
-    scanf("%d", &n);
-    a = (int *)malloc(n * sizeof(int));
-    for (i = 0; i < n; i++)
+    const int n = 10;
+    int *a = (int *)calloc(n, sizeof(int));
+
+    /* generate size random numbers from 0 to 100 */
+    for (int i = 0; i < n; i++)
     {
-        scanf("%d", &a[i]);
+        a[i] = rand() % 100;
     }
-
     merge_sort(a, n, 0, n - 1);
-    for (i = 0; i < n; i++)
+    for (int i = 0; i < n - 1; ++i)
     {
-        printf(" %d", a[i]);
+        assert(a[i] <= a[i + 1]);
     }
-
     free(a);
```

Comment:
```suggestion
    free(a);
    std::cout << "All tests have successfully passed!\n";
```

Reply:
I think you have mistaken this repo with c++ repo 😅 I'll change it to printf.